File: server/SpringRestTest/src/main/java/com/github/iblis/ms/SpringRestTest/repository/AuthorRepository.java
package com.github.iblis.ms.SpringRestTest.repository;

import com.github.iblis.ms.SpringRestTest.model.Author;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface AuthorRepository extends JpaRepository<Author, Long> {

    @Query("Select a From Author a")
    List<Author> getAuthors();

    @Query("Select a From Author a Where a.name Like %:namePart%")
    List<Author> getAuthorsLike(@Param("namePart") String namePart);

}

File: android/AndroidRestTestApp/app/src/main/java/com/example/androidresttestapp/repository/AuthorRepository.kt
package com.example.androidresttestapp.repository

import com.example.androidresttestapp.model.Author
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query

interface AuthorRepository {

    @GET("/authors")
    fun getAuthors() : Call<List<Author>>

    @GET("/find_authors")
    fun getAuthorsLike(@Query("like") like:String) : Call<List<Author>>
}